Changed the elif notation to else if

ekinbarut opened the pull request:
Changed the `elif` notation to `else if` per requirements in #74

ThakeeNathees commented on src/pk_compiler.c:
```diff
@@ -129,7 +129,7 @@ typedef enum {
   TK_FOR,        // for
   TK_IF,         // if
-  TK_ELIF,       // elif
+  TK_ELSEIF,     // else if
```
It should be `TK_ELSIF` and the keyword we're trying to rename to is `elsif`

ekinbarut replied:
I misunderstood the requirement, my bad. Changed the else if notation to elsif.

ThakeeNathees commented on src/pk_compiler.c:
```diff
@@ -190,7 +190,7 @@ static _Keyword _keywords[] = {
   { "for",      3, TK_FOR      },
   { "if",       2, TK_IF       },
-  { "elif",     4, TK_ELIF     },
+  { "else if",  4, TK_ELSEIF   },
```
it should be `"elsif"` and the 4 next to is the length of the keyword, which should be updated to 5 after the rename (and note that a keyword cannot contain a white space).

ekinbarut replied:
Changed the keyword count as well.

ThakeeNathees reviewed (changes requested):
Thank you for taking your time and effort to make this PR, and welcome to pocketlang.

I've left a review on the pr for some minor changes it requirs.

ekinbarut commented:
Changed the keyword notation from else if to elsif, removed the whitespace, and updated the keyword count.

ThakeeNathees commented on src/pk_compiler.c:
```diff
@@ -1078,7 +1078,7 @@ GrammarRule rules[] = {  // Prefix       Infix             Infix Precedence
   /* TK_FOR        */   NO_RULE,
   /* TK_IF         */   NO_RULE,
-  /* TK_ELIF       */   NO_RULE,
+  /* TK_ELSIF     */   NO_RULE,
```
Here the alignment is off by a white space, but that's okay, I'll merge it and fix it by myself. But it's in discussion at #74, I'd still like to hear from @xSavitar and @alexcpatel about the change and merge it by tomorrow.

Thanks!

ekinbarut replied:
Thanks for the review. Good luck!

xSavitar commented:
> Changed the `elif` notation to `else if` per requirements in #74

@ekinbarut, thanks for your change. It should be from `elif` to `elsif`. I mentioned a comment on the necessary task about the latter being very typo prone which is what I think your brain did here :]

Never the less, I think @ThakeeNathees, let's talk more about this a bit but I support the rename.

ThakeeNathees commented:
@ekinbarut I'm really happy to merge this PR, but could you rename all the pocketlang unit tests where we're using the `elif` keyword. Because merging this currently will fail the build status.

Thanks in advance!

(here is the list of `elif` we've used, it would be helpful for you to rename. `fmt=path : line`)
````
.tests\examples\brainfuck.pk : 52
.tests\examples\brainfuck.pk : 57
.tests\examples\brainfuck.pk : 62
.tests\examples\brainfuck.pk : 67
.tests\examples\brainfuck.pk : 70
.tests\examples\brainfuck.pk : 76
.tests\examples\brainfuck.pk : 83
.tests\examples\brainfuck.pk : 90
.tests\examples\brainfuck.pk : 97
.tests\examples\fizzbuzz.pk : 5
.tests\examples\fizzbuzz.pk : 6
.tests\lang\controlflow.pk : 10
.tests\lang\controlflow.pk : 15
.tests\lang\controlflow.pk : 16
.tests\lang\controlflow.pk : 21
.tests\lang\controlflow.pk : 24
.tests\lang\controlflow.pk : 42
.tests\lang\controlflow.pk : 44
```

ekinbarut commented:
Hello @ThakeeNathees, I renamed the keywords in the unit tests and saw that they were also used in prism and the 15-minute guide, so went ahead and changed those as well.

ThakeeNathees commented:
Thanks!

And we're glad to have you here, I'll open "good first issues" more If you're interested in this project and want to learn compiler principles together.
